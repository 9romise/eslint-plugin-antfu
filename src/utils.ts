import { ESLintUtils } from '@typescript-eslint/utils'

const hasDocs = [
  'consistent-list-newline',
  'if-newline',
  'import-dedupe',
  'top-level-function',
]

const blobUrl = 'https://github.com/antfu/eslint-plugin-antfu/blob/main/src/rules/'

export const createEslintRule = ESLintUtils.RuleCreator(
  ruleName => hasDocs.includes(ruleName)
    ? `${blobUrl}${ruleName}.md`
    : `${blobUrl}${ruleName}.test.ts`,
)
