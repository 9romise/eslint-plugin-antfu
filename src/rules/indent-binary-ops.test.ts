import { RuleTester } from '../../vendor/rule-tester/src/RuleTester'
import rule, { RULE_NAME } from './indent-binary-ops'

const valids = [
  '',
]
const invalids = [
`
if (
  a && (
    a.b ||
    a.c
  ) &&
  a.d
) {}
`,
`
const a = 
x +
   y * z
`,
`
if (
  aaaaaa >
bbbbb
) {}
`,
`
function foo() {
  if (a 
  || b 
      || c || d
        || (d && b)
  ) {
    foo()
  }
}
`,
]

const ruleTester: RuleTester = new RuleTester({
  parser: require.resolve('@typescript-eslint/parser'),
})

ruleTester.run(RULE_NAME, rule as any, {
  valid: valids,
  invalid: invalids.map(i => ({
    code: i,
    errors: null,
    onOutput: (output: string) => {
      expect(output).toMatchSnapshot()
    },
  })),
})
