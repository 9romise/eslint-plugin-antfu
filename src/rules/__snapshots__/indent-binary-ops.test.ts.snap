// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`indent-binary-ops > invalid > 
  function isNamedFunction(node:
  | Tree.ArrowFunctionExpression
  | Tree.FunctionDeclaration
  | Tree.FunctionExpression,
  ) {
    if (node.id)
      return true

    const parent = node.parent

    return parent.type === 'MethodDefinition'
              || (parent.type === 'Property'
                  && (
                    parent.kind === 'get'
                      || parent.kind === 'set'
                      || parent.method
                  )
              )
  }
 1`] = `
"
  function isNamedFunction(node:
  | Tree.ArrowFunctionExpression
  | Tree.FunctionDeclaration
  | Tree.FunctionExpression,
  ) {
    if (node.id)
      return true

    const parent = node.parent

    return parent.type === 'MethodDefinition'
      || (parent.type === 'Property'
              && (
                    parent.kind === 'get'
                    || parent.kind === 'set'
                      || parent.method
                  )
              )
  }
"
`;

exports[`indent-binary-ops > invalid > 
const a =
x +
   y * z
 1`] = `
"
const a =
x +
y * z
"
`;

exports[`indent-binary-ops > invalid > 
function foo() {
  if (a
  || b
      || c || d
        || (d && b)
  ) {
    foo()
  }
}
 1`] = `
"
function foo() {
  if (a
    || b
  || c || d
      || (d && b)
  ) {
    foo()
  }
}
"
`;

exports[`indent-binary-ops > invalid > 
if (
  a && (
    a.b ||
      a.c
  ) &&
    a.d
) {}
 1`] = `
"
if (
  a && (
    a.b ||
    a.c
  ) &&
  a.d
) {}
"
`;

exports[`indent-binary-ops > invalid > 
if (
  aaaaaa >
bbbbb
) {}
 1`] = `
"
if (
  aaaaaa >
  bbbbb
) {}
"
`;
